client/components/upgrade-nudge/index.jsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+import analytics from 'analytics';
+import sitesList from 'lib/sites-list';
+
+const sites = sitesList();
+
+export default React.createClass( {
+
+	displayName: 'UpgradeNudge',
+
+	propTypes: {
+		onClick: React.PropTypes.func,
+		className: React.PropTypes.string,
+		message: React.PropTypes.string,
+		icon: React.PropTypes.string,
+		event: React.PropTypes.string,
+		href: React.PropTypes.string,
+		jetpack: React.PropTypes.bool
+	},
+
+	getDefaultProps() {
+		return {
+			onClick: noop,
+			message: 'And get your own domain address.',
+			icon: 'star',
+			event: null,
+			jetpack: false
+		}
+	},
+
+	handleClick() {
+		if ( this.props.event ) {
+			analytics.tracks.recordEvent( 'calypso_upgrade_nudge_cta_click', {
```

Comment:
Some nudge ID / location would help in determining which nudge is best
But I think it can be summarized from url, still - would help